aderyn_core/src/detect/high/unchecked_return.rs

```diff
@@ -0,0 +1,116 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use crate::ast::{ASTNode, Expression, Identifier, MemberAccess, NodeID, NodeType};
+
+use crate::capture;
+use crate::context::browser::GetImmediateParent;
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct UncheckedReturnDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for UncheckedReturnDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        // When you have found an instance of the issue,
+        // use the following macro to add it to `found_instances`:
+        //
+        // capture!(self, context, item);
+
+        for function_call in context.function_calls() {
+            // Find the ID of FunctionDefinition that we're calling so that we may identify if there are returned params
+            match function_call.expression.as_ref() {
+                Expression::Identifier(Identifier {
+                    referenced_declaration: Some(id),
+                    ..
+                })
+                | Expression::MemberAccess(MemberAccess {
+                    referenced_declaration: Some(id),
+                    ..
+                }) => {
+                    if function_call
+                        .parent(context)
+                        .is_some_and(|node| node.node_type() == NodeType::Block)
```

Comment:
This is great! My first instinct would be to try to find all the possible positive cases, but you're using the sole negative case here! Great stuff 👍